src/Core/Services/Implementations/CipherService.cs

```diff
@@ -852,5 +906,75 @@ private void ValidateCipherLastKnownRevisionDateAsync(Cipher cipher, DateTime? l
                 );
             }
         }
+
+        private async Task DeleteAttachmentAsync(Cipher cipher, string attachmentId)
+        {
+            var data = cipher.GetAttachments()[attachmentId];
```

Comment:
Since this returns an `IDictionary<TKey, TValue>`, if the key, `attachmentId` doesn't exist, I'm pretty sure it will throw an exception (`KeyNotFoundException` if memory serves). Is this acceptable behavior in this instance or should this condition be handled more gracefully? I also don't see a "not found" path through here as well, seems like it assumes the attachment exists all the way through.

Reply:
You're right. This is called as a cleanup method. It needs to work on each the repository, the cipher, and the file storage service, but it shouldn't complain in the cipher doesn't know about the attachment.

I'm adjusting this to take the Attachment Metadata as a parameter and try to delete from all three no matter what. Each of the respective delete methods correctly handle a not found case.